sql/sql_class.cc

```diff
@@ -3308,6 +3308,12 @@ bool Query_result_send::send_result_set_metadata(List<Item> &list, uint flags)
     /* Metadata is already send during first try that failed so avoid
     resending it. Just plan to send the result. */
     WSREP_DEBUG("Skip resending metadata if query is being re-tried");
+
+    /* Initialization is required for proper asserting */
+    if (thd->get_protocol()->start_result_metadata(
```

Comment:
Is this OK? `start_result_metadata` returns false in case of success.
So `is_result_set_started` will not be set in success path.

Reply:
No, it's wrong, fixed.